### PR TITLE
ENH: Add intent codes used by FSL to avoid future clashes

### LINKED ISSUE #66: Reserve or codify FSL intent codes
Just want to flag that FSL has custom intent codes for [FNIRT](https://github.com/mshvartsman/FSL/blob/7aa2932949129f5c61af912ea677d4dbda843895/src/warpfns/fnirt_file_reader.h#L100-L107) and [TOPUP](https://github.com/mshvartsman/FSL/blob/fsl-5.0.9/src/topup/topup_file_io.h#L100-L106) files.

Because these exist in the wild, we accept them in nibabel (nipy/nibabel#528; [nibabel/nifti1.py#L239-L251](https://github.com/nipy/nibabel/blob/ed3cdb5a33fdf1dd49e037600927f3b213f61f08/nibabel/nifti1.py#L239-L251)).

I don't have a strong opinion on whether these should be adopted in the official standard, but they should probably be reserved with a note, so a clash doesn't occur in the future.

cc @pauldmccarthy

### PR DESCRIPTION
**Pull request by effigies**
These intent codes were present in the FSL 5.0.9 source code. I have not downloaded more recent versions to see if they have added new codes.

@pauldmccarthy I figure someone from the FSL world should probably have a look over this.

Closes #66.

### CODE REVIEW

**Comment by pauldmccarthy** [comment]
Hey @effigies, sorry for the delay - these are all the FSL-specific codes that I am aware of. 

The  `_COEFFICIENTS` file types also do non-standard things with the sform, qform, and pixdim fields (to store extra information needed to reconstruct a full displacement field from the coefficients), but I'm not sure if that's something that this library (or `nibabel`) should be responsible for handling?

**Comment by hjmjohnson** [comment]
@effigies @pauldmccarthy I would prefer to have the codes listed in the standard with a comment that implementation of encode/decode of images with this intent is OPTIONAL, and not explicitly supported by the standard.

File: nifti2/nifti1.h [hunk 1]
```diff
@@ -957,6 +957,21 @@ typedef struct { unsigned char r,g,b; } rgb_byte ;
 
 #define NIFTI_INTENT_SHAPE        2005
 
+ /*! The following intent codes have been used by FSL FNIRT for
+     displacement/coefficient files.  */
```
Comment (hjmjohnson) [inline nifti2/nifti1.h]: @effigies Discussion that these intent flags support community supported extensions to NIFTI, but implementing the encode/decode behaviors of these intents is not supported by the NIFTI standard.

**Comment by effigies** [comment]
@hjmjohnson Thanks for the review. Added the following boilerplate above each of the two groups of intents:

```
     These codes are included to prevent clashes in community-created
     extensions to NIfTI. Encoding and decoding behavior for these
     intents is not specified by the standard, and support is OPTIONAL
     for conforming implementations.
```

Let me know if you prefer other wording/organization.

**Comment by effigies** [comment]
@pauldmccarthy I would say that it's mostly the responsibility of the library to provide reliable representations of the encoded structures so that the application can choose how to interpret them.

**Pull request merged**